### PR TITLE
fix: ensure production image is built with VITE ENV vars

### LINKED ISSUE #38: Defective production docker image
## Expected Behavior

When building an image it should run and successfully interact with ArgosAPI and ArgosFileManagerAPI

## Current Behavior

When running the produced image, it cannot perform GET or POST queries to other services.

## Steps to Reproduce

1. Build a new docker image `docker compose -f docker-compose-prod.yml build`
2. Run the produced image `docker compose -f docker-compose-prod.yml up -d``
3. Run the other services (argos-api and argos-filiemanager)
4. Attempt to upload a new project

## Context (Environment)

It makes the deployment of this repository near impossible because all the features related with the APIs are not working.

## Detailed Description

I propose update the `Dockerfile.prod` file to use env vars so that when building the project vite will process them.

### PR DESCRIPTION
**Pull request by zrock-dev**
fixes #38
* * * *

## What's in this pull request?

The image is being built without the VITE ENV vars which result in a
defective image that cannot communicate with other services.

- Add a validation to ensure ENV vars are present.
- Update prod compose file with ENV build arguments.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

1. Build and run a new image `docker compose -f docker-compose-prod.yml up -d --build`
2. Run argos-api and argos-filelamanger services
3. Load a new project using the UI

Make sure to use production environmental variables:
```
VITE_FILE_MANAGER_API_BASE_URL=http://prod-argos.westus2.cloudapp.azure.com:8080
VITE_ARGOS_API_BASE_URL=http://prod-argos.westus2.cloudapp.azure.com:8081
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: docker-compose-prod.yml [hunk 1]
```diff
@@ -4,8 +4,11 @@ services:
     build:
       context: .
       dockerfile: Dockerfile.prod
+      args:
+        VITE_ARGOS_API_BASE_URL: ${VITE_ARGOS_API_BASE_URL}
+        VITE_FILE_MANAGER_API_BASE_URL: ${VITE_FILE_MANAGER_API_BASE_URL}
     ports:
-      - "5175:80"
+      - "5173:80"
```
Comment (ronaldo-mendoza-jalafund) [inline docker-compose-prod.yml]: This port is using develop

**Review by ronaldo-mendoza-jalafund — CHANGES_REQUESTED** [review]
Resolves comments

File: docker-compose-prod.yml (hunk 1 above)
Comment (zrock-dev) [reply docker-compose-prod.yml]: Okay I've returned it to its original value

**Review by ronaldo-mendoza-jalafund — APPROVED** [review]
Nice job

**Review by ignacio-villarroel-jalafund — APPROVED** [review]
LGTM

**Comment by ignacio-villarroel-jalafund** [comment]
Screenshot from 2024-12-07 16-57-49
Project runs without any issue.

**Pull request merged**